test/widgets/inset_shadow.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:ui' as ui;
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:legacy_checks/legacy_checks.dart';
+import 'package:zulip/widgets/inset_shadow.dart';
+
+import '../flutter_checks.dart';
+
+void main() {
+  testWidgets('no additional padding added', (tester) async {
+    await tester.pumpWidget(const Directionality(
+      textDirection: TextDirection.ltr,
+      child: Align(
+        // Position child at the top-left corner of the box at (0, 0)
+        // to ease the check on [Rect] later.
+        alignment: Alignment.topLeft,
+        child: SizedBox(width: 20, height: 20,
+          child: InsetShadowBox(top: 7, bottom: 3,
+            color: Colors.red,
+            child: Placeholder())))));
+
+    final childRect = tester.getRect(find.byType(Placeholder));
+    check(childRect).equals(const Rect.fromLTRB(0, 0, 20, 20));
```

Comment:
I believe this check would pass no matter what `InsetShadowBox` attempted to do for its layout. Its parent (the `SizedBox`) has decreed exactly what its size is, so it's not able to be any other size. See:
https://docs.flutter.dev/get-started/fundamentals/layout#constraints
https://api.flutter.dev/flutter/rendering/RenderBox-class.html#layout

It's worth trying to verify that — edit InsetShadowBox so that it does use `top` and `bottom` as padding, and see if this test still passes. I believe it will.

To make the test effective, I think what's needed is to have a `SizedBox` as the child, not the parent. Then do confirm that that test fails if you make that change to InsetShadowBox.

Reply:
We actually caught a bug with the implementation that allows `InsetShadowBox` to modify the constraints from the parent. The test is updated to catch layout changes.